tests/poll.rs

```diff
@@ -310,6 +311,67 @@ fn registry_operations_are_thread_safe() {
     handle3.join().unwrap();
 }
 
+#[test]
+fn poll_ok_after_cancelling_pending_ops() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+    let address = listener.local_addr().unwrap();
+
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+    let registry1 = Arc::clone(&registry);
+
+    let barrier = Arc::new(Barrier::new(2));
+    let barrier1 = Arc::clone(&barrier);
+
+    registry
+        .register(&mut listener, ID1, Interest::READABLE)
+        .unwrap();
+
+    // Call a dummy poll just to submit an afd poll request
+    poll.poll(&mut events, Some(Duration::from_millis(0)))
+        .unwrap();
+
+    // this reregister will cancel the previous pending poll op
+    registry
+        .reregister(&mut listener, ID1, Interest::READABLE | Interest::WRITABLE)
```

Comment:
`WRITABLE` doesn't make sense.

Reply:
I know, I just use it to be have something different in the `reregister` call. While not having a practical usage, it helps during debugging to have different flags set on reregister. I'd rather keep it than remove it.

Reply:
I added a comment with its purpose, but if you have a strong opinion against it I'll remove it.

Reply:
I often look at tests of a crate as example usage. So if we have invalid code in our tests we set a bad precedent that's why I like to avoid it.

Reply:
good point, will remove writable